services/core/MasterDriverAgent/master_driver/agent.py

```diff
@@ -400,12 +427,166 @@ def heart_beat(self):
             
     @RPC.export
     def revert_point(self, path, point_name, **kwargs):
-        self.instances[path].revert_point(point_name, **kwargs)
-    
+        if path in self._override_devices:
+            raise OverrideError(
+                "Cannot revert point on device {} since global override is set".format(path))
+        else:
+            self.instances[path].revert_point(point_name, **kwargs)
+
     @RPC.export
     def revert_device(self, path, **kwargs):
-        self.instances[path].revert_all(**kwargs)
+        if path in self._override_devices:
+            raise OverrideError(
+                "Cannot revert device {} since global override is set".format(path))
+        else:
+            self.instances[path].revert_all(**kwargs)
 
+    # Turn on override condition on all the devices matching the pattern.
+    @RPC.export
+    def set_override_on(self, pattern, duration=1.0, indefinite_duration=False, failsafe_revert=True, staggered_revert=False):
+        self._set_override_on(pattern, duration, indefinite_duration, failsafe_revert, staggered_revert)
+
+    def _set_override_on(self, pattern, duration=1.0, indefinite_duration=False, failsafe_revert=True, staggered_revert=False):
```

Comment:
Could we use duration <= 0 to indicate infinite duration and make that the default?